### PR TITLE
Session tracking on by default

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Enable automatic session tracking by default in React Native.

## Changeset

- Updates Cocoa to the current state of the pre-release next branch, and Android to v4.5.0.
- Slight alterations to the logic for passing the notify/session endpoints to the native layer
- JS Configuration is now true by default

## Tests

One manual test on both Cocoa/Android to ensure that a session is sent by default.

### Outstanding Questions

- Will need updating to the released version of Cocoa.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

**Comment by fractalwrench** [comment]
One potential issue I can see is that we currently advise to initialise the notifier in the following ways:

```
[BugsnagReactNative start];
BugsnagReactNative.start(this);
```

This instantiates a bugsnag client immediately, which as part of construction, sends any stored sessions immediately to sessions.bugsnag.com by default. An alternative would be to document that this is the case, and advise that the existing `BugsnagReactNative.start(this, config);` option is used for both clients instead, in the case of on-prem.

Also, as noted in the description, this will require some more manual testing once the Cocoa notifier has been fully released.

File: android/src/main/java/com/bugsnag/BugsnagReactNative.java [hunk 1]
```diff
@@ -213,15 +213,14 @@ private void configureRuntimeOptions(Client client, ReadableMap options) {
...
-          String endpoint = options.getString("sessionsEndpoint");
-          if (endpoint != null && endpoint.length() > 0)
-              config.setSessionEndpoint(endpoint);
+          if (notify != null && notify.length() > 0) {
+              if (options.hasKey("sessionsEndpoint")) {
+                  String sessions = options.getString("sessionsEndpoint");
+                  config.setEndpoints(notify, sessions);
+              }
+          }
       }
```
Comment (kattrali) [inline android/src/main/java/com/bugsnag/BugsnagReactNative.java]: There should be a log message here if only `sessionsEndpoint` is set

**Comment by kattrali** [comment]
> An alternative would be to document that this is the case, and advise that the existing `BugsnagReactNative.start(this, config);` option is used for both clients instead, in the case of on-prem.

Agreed, this should also happen in case there's a crash between native and JS initialization times.

File: android/src/main/java/com/bugsnag/BugsnagReactNative.java (hunk 1 above)
Comment (fractalwrench) [reply android/src/main/java/com/bugsnag/BugsnagReactNative.java]: A warning is logged by bugsnag-android, if that's sufficient? https://github.com/bugsnag/bugsnag-android/blob/master/sdk/src/main/java/com/bugsnag/android/Configuration.java#L160

Comment (kattrali) [reply android/src/main/java/com/bugsnag/BugsnagReactNative.java]: In this case it would never get to the bugsnag-android warning since `setEndpoints` is only called if both are set.
